apm/src/app/movies/movies.service.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { catchError, Observable, tap, throwError } from 'rxjs';
+import { IMoviesResult } from './movies';
+
+@Injectable()
+export class MoviesService {
+
+  constructor(private readonly http: HttpClient) { }
+
+  public getMovies(): Observable<IMoviesResult>  {
```

Comment:
I think  this method should be returning `Observable<IMovies[]>` instead, because all this method should do it return a list of movies as the name suggests.

So you can use the `map` operator in Rxjs to transform the incoming response into something else